openapi/openapi/src/main/scala/endpoints/openapi/model/OpenApi.scala

```diff
@@ -269,65 +304,467 @@ object OpenApi {
 
 }
 
-case class Info(
-    title: String,
-    version: String
-)
+class Info private (
+    val title: String,
+    val version: String
+) extends Serializable {
 
-case class PathItem(
-    operations: Map[String, Operation]
-)
+  override def toString: String =
+    s"Info($title, $version)"
 
-case class Components(
-    schemas: Map[String, Schema],
-    securitySchemes: Map[String, SecurityScheme]
-)
+  override def equals(other: Any): Boolean =
+    other match {
+      case that: Info => title == that.title && version == that.version
+      case _          => false
+    }
 
-case class Operation(
-    summary: Option[String],
-    description: Option[String],
-    parameters: List[Parameter],
-    requestBody: Option[RequestBody],
-    responses: Map[String, Response],
-    tags: List[String],
-    security: List[SecurityRequirement],
-    callbacks: Map[String, Map[String, PathItem]],
-    deprecated: Boolean
-)
+  override def hashCode(): Int =
+    Hashing.hash(title, version)
 
-case class SecurityRequirement(
-    name: String,
-    scheme: SecurityScheme,
-    scopes: List[String] = Nil
-)
+  private[this] def copy(
+      title: String = title,
+      version: String = version
+  ): Info =
+    new Info(title, version)
 
-case class RequestBody(
-    description: Option[String],
-    content: Map[String, MediaType]
+  def withTitle(title: String): Info =
+    copy(title = title)
+
+  def withVersion(version: String): Info =
+    copy(version = version)
+
+}
+
+object Info {
+
+  def apply(title: String, version: String): Info =
+    new Info(title, version)
+
+}
+
+class PathItem(
```

Comment:
Should we make the constructor `private`?

Reply:
Yes, I think that makes sense.